calyx/src/passes/group_to_seq.rs

```diff
@@ -0,0 +1,258 @@
+use crate::analysis::OrderAnalysis;
```

Comment:
As much as possible, try not to implement things from scratch. For example, the `Group` object provides a bunch of API functions to manage done conditions: https://docs.calyxir.org/source/calyx/ir/struct.Group.html#method.done_cond